### DISCUSSION TITLE
How to handle a non-persistent home directory?

### DISCUSSION THREAD

**Discussion by coffeepenbit** [twpayne/chezmoi#3347] [Q&A] [answered]
### Problem
I'm trying to deal with a non-persistent home directory on one of my machines. The home directory is recreated each startup.

### Solution attempts
I've tried storing the destination files pertinent to chezmoi in a persistent location and symlinking, which results in errors (I believe this is a documented as out-of-scope for chezmoi).

I've also considered having `chezmoi apply` automatically run at startup, but then I run into the issue of dealing with my secrets manager requesting a password.

The only other solution I can think of is doing something convoluted using chezmoiscripts. **I'd like to confirm I'm not missing something here before I go through the work of setting this up.**

**Comment by twpayne** [comment]
I'd suggest `chezmoi apply` as the best way.

Another option is to use `chezmoi archive` to generate an archive of your dotfiles and then unpack that on startup. This will avoid you having to type in your secrets manager password each time.

**Reply by coffeepenbit** [reply]
Thanks for the quick reply!

The archive approach seems nice for this; however, I tried tried adding the command in a `run_after_` script. Chezmoi then warns me about a modified config, as well as prompting for the same password I'm trying to avoid dealing with.

I'm guessing this is something wrong on my side because I can't clean anything in the source code that would cause this by executing the archive command. I'll keep hacking away at this.

**Reply by halostatue** [reply]
You shouldn't be able to run `chezmoi archive` in a `run_after_` script because it is a specialized form of `chezmoi apply` (apply into a tarball), so it needs to work with state.

You *might* have some luck with a variation of @Lockszmith-GH's suggestions in #2574 (two different states; one with secrets the other without) or you have a workflow such that on home creation, you unpack the previously stored archive and when you know there are updates, you have a script that applies and then makes a new archive in the persistent store.

**Reply by coffeepenbit** [reply, marked as answer]
Thanks for the insight @halostatue and for confirming the behavior I observed.

I took inspiration from the previous approach and ran archived the files from `chezmoi managed` inside a `run_after_` script. This will always have the latest files and I won't need to enter the password (after running `chezmoi apply`

Thanks for the support, you two
